src/core/util.c

```diff
@@ -607,8 +607,14 @@ JanetBinding janet_resolve_ext(JanetTable *env, const uint8_t *sym) {
     };
 
     /* Check environment for entry */
-    if (!janet_checktype(entry, JANET_TABLE))
-        return binding;
+    if (!janet_checktype(entry, JANET_TABLE)) {
+        Janet fn_entry = janet_table_get(env, janet_ckeywordv("missing-symbol"));
+        if (janet_checktype(fn_entry, JANET_FUNCTION)) {
+            Janet args[2] = { janet_wrap_symbol(sym), janet_wrap_table(env) };
+            entry = janet_call(janet_unwrap_function(fn_entry), 2, args);
+        }
+        if (!janet_checktype(entry, JANET_TABLE)) return binding;
+    }
```

Comment:
So another issue with using janet_call is that currently, janet_compile will not panic. This is by design so that we can always easily get compilation errors. Instead, do what macro expansion does and use janet_pcall.

Reply:
If I'm following the code correctly, [`macroexpand1`](https://github.com/janet-lang/janet/blob/f2f421a0a25aa2fd850372d9c160eac6bb3b8fe3/src/core/compile.c#L664-L699) creates a fiber to execute the function manually. I just want to confirm, would you prefer me to do it that way or is calling `janet_pcall` fine?

Reply:
Ah, yeah they are quite similar but creating the fiber manually is fine. Just we shouldn't panic and instead issue a normal compilation error. Perhaps the compiler could be updated to use the usual panic path, but for now we should instead report errors with janetc_error.

Reply:
I think the way that I'd originally implemented this wasn't correct. I put the lookup in `janet_resolve_ext()` rather than in `janetc_resolve()`. That didn't allow for the compiler to look up the symbol and output appropriate errors if there was a problem with the lookup handler.

To fix that, I've moved the missing symbol lookup into `compile.c` and added errors analogous to those that are output during macro expansion. Let me know if this isn't what you had in mind.

Reply:
That looks good so far, yes. Essentially we just need to make sure that the `compile` function always outputs good errors. If it doesn't, you can get stuff that is _really_ hard to debug.